tools/check_links.py

```diff
@@ -18,13 +18,13 @@
 
 for doc in docs:
 	with open(doc) as f:
-		data = f.read()
-		r = link_re.findall(data)
-		for link in r:
-			links += [(doc, link)]
+		lines = f.readlines()
+		for line in range(len(lines)):
+			for match in link_re.finditer(lines[line]):
+				links += [(doc, match.group(1), line + 1, match.start(1))]
```

Comment:
In python something like this can be done with `enumerate`:
```
for i, line in enumerate(lines):
	for match in link_re.finditer(line):
		links += [(doc, match.group(1), i + 1, match.start(1))]
```

Reply:
Done